src/toast/_libtoast/tod_filter.cpp

```diff
@@ -125,6 +125,48 @@ void init_tod_filter(py::module & m) {
             None.
     )");
 
+    m.def("fourier",
+          [](py::buffer angle, py::buffer templates, size_t start_order,
+             size_t stop_order) {
+              pybuffer_check_1D <double> (angle);
+              py::buffer_info info_angle = angle.request();
+              py::buffer_info info_templates = templates.request();
```

Comment:
I know we need to refactor the existing compiled code to move things directly into the `_libtoast` extension...  For consideration when writing new functions, there is a new templated helper function for extracting the raw pointers from a  `py::buffer` while simultaneously checking optional shape constraints.  See for example:  https://github.com/hpc4cmb/toast/blob/3252edf59d75a76adbc93f6947c1ab1bbb6fa3f9/src/toast/_libtoast/ops_mapmaker_utils.cpp#L120-L170

So this function would look like:
```
std::vector <int64_t> temp_shape(2);

double * pangle = extract_buffer <double> (angle, "angle", 1, temp_shape, {-1});
size_t nsample = temp_shape[0];

double * ptemplates = extract_buffer <double> (templates, "templates", 2, temp_shape, {-1, nsample});
size_t ntemplate = temp_shape[0];

if (ntemplate != 2 * (stop_order - start_order)) {
...
```
And the `extract_buffer` function raises an error if the shape constraints of the input buffer are not met.

Reply:
I agree, that is more elegant